[ROCm] move nccl stream to member of StreamGroup

jeffdaily opened the pull request:
This allows the compute and nccl stream to be force-initialized as immediate
siblings which is necessary for ROCm performance.

jeffdaily committed:
move nccl stream to member of StreamGroup

This allows the compute and nccl stream to be force-initialized as immediate
siblings which is necessary for ROCm performance.

rthadur commented:
@jeffdaily can you please resolve conflicts ?

jeffdaily committed:
Merge branch 'master' into google_upstream_nccl_stream_early

mellanox-github commented:
Can one of the admins verify this patch?

rthadur commented:
@jeffdaily can you please resolve conflicts and we can run the tests once you are done, thank you

jeffdaily committed:
Merge branch 'master' into google_upstream_nccl_stream_early

jeffdaily committed:
early nccl stream is now specific to ROCm platform

This also disables the multi-node emulation of the nccl_manager_test
for the ROCm platform.
